File: hw/chicaiza/u1/hw9AntSimulator/AntSimulator/src/model/Nest.java
package model;

public class Nest {
    private Cell location;
    private int foodAmount;

    public Nest(Cell location) {
        this.location = location;
        this.foodAmount = 0;
        // Initialize the nest at a specific cell with no food
    }

    // Add food to the nest
    public void addFood(int mg) {}

    // Create a new ant in the nest
    public void createAnt() {}
}

File: hw/bonilla/u1/HW09AntSimulator/Nest.js
import { Food } from './Food.js';
import { Ant } from './Ant.js';

export class Nest {
  constructor(cell) {
    this.position = { x: cell.x, y: cell.y };
    this.cell = cell;
    this.foodStock = 0;
    cell.nest = this;
  }

  addFood(food) {
    this.foodStock += food.amount;
  }

  removeFood(amount) {
    const used = Math.min(amount, this.foodStock);
    this.foodStock -= used;
    return new Food(used);
  }

  canCreateAnt() {
    return this.foodStock >= 5;
  }

  createAnt(colony) {
    if (!this.canCreateAnt()) return null;
    this.foodStock -= 5;
    const ant = new Ant(this.cell, colony);
    colony.addAnt(ant);
    return ant;
  }
}